noir-projects/aztec-nr/aztec/src/messages/encryption/aes128.nr

```diff
@@ -45,17 +90,24 @@ fn extract_close_to_uniformly_random_256_bits_from_ecdh_shared_secret_using_pose
         [shared_secret.x, shared_secret.y],
         GENERATOR_INDEX__SYMMETRIC_KEY_2,
     );
-    let rand1_bytes: [u8; 16] = rand1.to_le_bytes();
-    let rand2_bytes: [u8; 16] = rand2.to_le_bytes();
+
+    let rand1_bytes: [u8; 32] = rand1.to_be_bytes();
+    let rand2_bytes: [u8; 32] = rand2.to_be_bytes();
+
     let mut bytes: [u8; 32] = [0; 32];
     for i in 0..16 {
-        bytes[i] = rand1_bytes[i];
-        bytes[i + 1] = rand2_bytes[i];
+        // We take bytes from the "little end" of the be-bytes arrays:
+        let j = 32 - i - 1;
+        bytes[i] = rand1_bytes[j];
+        bytes[16 + i] = rand2_bytes[j];
     }
+
     bytes
 }
 
-// TODO(#10537): Consider nuking this function.
+// We'll keep this function around, because some users might need it if their
+// shared_secret is derived in a way which is not safe for usage with the
+// poseidon2 approach.
 fn extract_close_to_uniformly_random_256_bits_from_ecdh_shared_secret_using_sha256(
```

Comment:
We're not using it though are we?

Reply:
No, we won't be using it if this PR goes through.

Reply:
If you'd like to remove the sha256 stuff, I'll do it as a separate PR in future :)

Reply:
Yea i'd remove unused things.

Reply:
Removed